cylc/uiserver/resolvers.py

```diff
@@ -205,30 +219,32 @@ async def clean(
         executor: 'Executor',
         log: 'Logger'
     ):
-        """Calls `init_clean`"""
+        """Calls `cylc clean`"""
         # Convert Schema options → cylc.flow.workflow_files.init_clean opts:
         opts = _schema_opts_to_api_opts(args, schema=CleanOptions)
-        # Hard set remote timeout.
-        opts.remote_timeout = "600"
+        opts.remote_timeout = "600"  # Hard set remote timeout.
+        opts.skip_interactive = True  # disable interactive prompts
 
-        # clean each requested flow
-        for tokens in workflows:
-            log.info(f'Cleaning {tokens}')
-            workflow = tokens.pop('workflow')
-            try:
-                await asyncio.wrap_future(
-                    executor.submit(init_clean, workflow, opts)
-                )
-            except Exception as exc:
-                if isinstance(exc, ServiceFileError):  # Expected error
-                    # The "workflow still running" msg is very long
-                    msg = str(exc).split('\n')[0]
-                elif isinstance(exc, WorkflowFilesError):  # Expected error
-                    msg = str(exc)
-                else:  # Unexpected error
-                    msg = f"{type(exc).__name__}: {exc}"
-                    log.exception(msg)
-                return cls._error(msg)
+        # Convert tokens into string IDs
+        workflow_ids = [tokens.workflow_id for tokens in workflows]
+
+        # run cylc clean
+        log.info(f'Cleaning {" ".join(workflow_ids)}')
+        try:
+            await asyncio.get_event_loop().run_in_executor(
+                executor,
+                partial(_clean, workflow_ids, opts=opts),
```

Comment:
No need to involve functools
```suggestion
                executor, _clean, workflow_ids, opts
```